pwiz_tools/Skyline/SkylineFiles.cs

```diff
@@ -3344,6 +3344,11 @@ public void ShowPublishDlg(IPanoramaPublishClient publishClient)
             {
                 return false;
             }
+            catch (Exception e)
+            {
+                MessageDlg.ShowWithException(Program.MainWindow, TextUtil.LineSeparate(Resources.WebPanoramaPublishClient_SupportedVersionsJson_Error_parsing_JSON_data, e.Message), e);

```

Comment:
You should use "this" instead of "Program.MainWindow".
The error message should be less specific since at the spot where we are catching we don't know what might have caused it. I see that there is already a string in Resources.resx which you can use: Resources.RemoteSession_FetchContents_There_was_an_error_communicating_with_the_server__ which just says "There was an error communicating with the server: "